### ISSUE TITLE
SPFX Custom Search extensibility library app installation fails

### ISSUE THREAD

**Issue by arunakshi** [SharePoint/sp-dev-docs#5303]
## Category
[x] Bug

## Environment details
SharePoint Online
SPFX v1.9.1
 
We developed an search extensibility library based on the Pnp demo Project. We want to deploy to tenant wide but not installing to all sites and site Collections. We are trying to install the app per site base. At the moment it is not working when we try to install our custom search extensiblity library app in site collection. I am not able to see Errors when i try to check the Installation Errors. 

It is mentioned in the documentation that Library components are not supported whe solution is deployed using the site Collection app catalog in the link: https://docs.microsoft.com/en-us/sharepoint/dev/spfx/library-component-overview. I can still deploy to site Collection app catalog. 

For our solution, we need the tenant wide deployment and our search library app needs to be added to specific sites manually. This is failing. 

Can anyone please help on this issue.

**Comment by arunakshi** [comment]
A short update on this, I am having this issue when I take Pnp Search extensibility library Project. I did no changes for the project and just compiled and packaged it and try to add to Site Collection. Got same error!!!

**Comment by andrewconnell** [comment]
@arunakshi said:
> I am having this issue when I take Pnp Search extensibility library demo Project

What demo project are you referring to?

**Comment by arunakshi** [comment]
@andrewconnell, I took the demo project from PnP Modern Search solution. Here is the link to Modern Search solution https://github.com/microsoft-search/pnp-modern-search/. In the solution, there is a folder called search-extensibility-library. I took that folder as my Search Extensibility Library project to do my enhancements. For the enhancements, I followed the documentation from here: https://microsoft-search.github.io/pnp-modern-search/search-extensibility-library/getting-started/
